services/supervisor/src/api.rs

```diff
@@ -12,6 +12,578 @@
+//!     .create();
+//!
+//! let pending_proposal: Option<ConfigProposalWithHash> = testkit
+//!     .api()
+//!     .public(ApiKind::Service("supervisor"))
+//!     .get("config-proposal")?;
+//!
+//! // Will be none, since we did not send a proposal.
+//! assert!(pending_proposal.is_none());
+//! # Ok(())
+//! # }
+//! ```
+//!
+//! # Private API
+//!
+//! ## Request to Deploy an Artifact
+//!
+//! | Property    | Value |
+//! |-------------|-------|
+//! | Path        | `/api/supervisor/v1/deploy-artifact` |
+//! | Method      | POST   |
+//! | Query type  | [`DeployRequest`] |
+//! | Return type | [`Hash`] |
+//!
+//! Requests the deployment of a certain artifact.
+//!
+//! Depending on the supervisor operating mode, it may be required to send such a request to
+//! majority of nodes (in "decentralized" mode), or one request will be enough (in "simple" mode).
+//!
+//! After receiving a deployment request, supervisor creates a corresponding transaction, signs it
+//! with node's keys and broadcasts the transaction within the network. The hash of broadcast
+//! transaction is returned from the endpoint.
+//!
+//! For more details on deploy requests, see [crate documentation](../index.html).
+//!
+//! **Warning:** `DeployRequest` structure should be serialized using corresponding protobuf message,
+//! and represented as a hexadecimal string.
+//!
+//! [`DeployRequest`]: ../struct.DeployRequest.html
+//! [`Hash`]: https://docs.rs/exonum-crypto/latest/exonum_crypto/struct.Hash.html
+//!
+//! ```
+//! use exonum::{crypto::Hash, helpers::Height, merkledb::BinaryValue};
+//! use exonum_rust_runtime::ServiceFactory;
+//! use exonum_supervisor::{DeployRequest, Supervisor};
+//! use exonum_testkit::{ApiKind, TestKitBuilder};
+//!
+//! # use exonum_derive::*;
+//! # use exonum_rust_runtime::Service;
+//! #
+//! # #[derive(Debug, ServiceFactory, ServiceDispatcher)]
+//! # #[service_factory(artifact_name = "exonum.doc.SomeService", artifact_version = "0.1.0")]
+//! # pub struct SomeService;
+//! #
+//! # impl Service for SomeService {}
+//! #
+//! # fn config_for_service() -> Vec<u8> {
+//! #     Vec::new()
+//! # }
+//! #
+//! # fn main() -> Result<(), failure::Error> {
+//! let mut testkit = TestKitBuilder::validator()
+//!     .with_rust_service(Supervisor)
+//!     .with_artifact(Supervisor.artifact_id())
+//!     .with_instance(Supervisor::simple())
+//!     .with_rust_service(SomeService)
+//!     .create();
+//!
+//! // In this example, we will try to deploy supervisor service.
+//! let deploy_request = DeployRequest {
+//!     artifact: SomeService.artifact_id(),
+//!     spec: config_for_service(),
+//!     deadline_height: Height(10),
+//! };
+//!
+//! let tx_hash: Hash = testkit
+//!     .api()
+//!     .private(ApiKind::Service("supervisor"))
+//!     .query(&deploy_request)
+//!     .post("deploy-artifact")?;
+//!
+//! let block = testkit.create_block();
+//! let result = block[tx_hash].status();
```

Comment:
Nit: Could this be shortened to

```rust
block[tx_hash].status()?;
```

Reply:
I thought that with `assert` the example will be more demonstrative (in other places `?` is used to not-so-important determine places, where we don't want to `unwrap`, but here it's the key point of the example).